README.md

```diff
@@ -45,18 +44,97 @@ Check out our live demo on [demo.pyrra.dev](https://demo.pyrra.dev)!
 
 Feel free to give it a try there!
 
-## Installation
+## How It Works
+
+There are three components of Pyrra, all of which work through a single binary:
+
+- The frontend which displays SLOs, error budgets, burn rates, etc.
+- The api which delivers information about SLOs from a backend (like Kubernetes)
+  to the frontend.
+- A reconciler which watches for new SLO objects and then creates Prometheus 
+  recording rules for each.
+
+For the reconciler to do its magic, an SLO object needs to provided in 
+YAML-format:
+
+```yaml
+apiVersion: pyrra.dev/v1alpha1
+kind: ServiceLevelObjective
+metadata:
+  name: pyrra-api-errors
+  namespace: monitoring
+  labels:
+    prometheus: k8s
+    role: alert-rules
+spec:
+  target: '99'
+  window: 2w
+  description: Pyrra's API requests and response errors over time. For now, this SLO is a comined SLO for all endpoints.
+  indicator:
+    ratio:
+      errors:
+        metric: http_requests_total{job="pyrra",code=~"5.."}
+      total:
+        metric: http_requests_total{job="pyrra"}
+      grouping:
+        - route
+```
+
+Depending on your mode of operation, this information is provided through an
+object in Kubernetes, or read from a static file.
+
+In order to calculate error budget burn rates, Pyrra will then proceed to create
+[Prometheus recording rules](https://prometheus.io/docs/prometheus/latest/configuration/recording_rules/#recording-rules)
+for each SLO.
+
+The following rules would be created for the above example:
+
+```
+pyrra-api-errors:increase2w
+pyrra-api-errors:burnrate3m
+pyrra-api-errors:burnrate15m
+pyrra-api-errors:burnrate30m
+pyrra-api-errors:burnrate1h
+pyrra-api-errors:burnrate3h
+pyrra-api-errors:burnrate12h
+pyrra-api-errors:burnrate2d
```

Comment:
```suggestion
http_requests:increase2w
http_requests:burnrate3m
http_requests:burnrate15m
http_requests:burnrate30m
http_requests:burnrate1h
http_requests:burnrate3h
http_requests:burnrate12h
http_requests:burnrate2d

The recording rules names are based on the originally provided metric. 
The recording rules contain the necessary labels to uniquely identify the recording rules in case there are multiple ones available.
```